src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int DEFAULT_ARRAY_LENGTH = 16;
+    private int size;
+    private Node<K, V> [] values;
+
+    public MyHashMap() {
+        values = new Node[DEFAULT_ARRAY_LENGTH];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        if (values[getIndexFromKey(key)] == null) {
+            Node<K, V> newNode = new Node<>(key, value, null);
+            values[getIndexFromKey(key)] = newNode;
+            size++;
+        } else {
+            Node<K, V> currentNode = values[getIndexFromKey(key)];
+            while (currentNode != null) {
+                if (currentNode.nodeKey != null && currentNode.nodeKey.equals(key)
+                        || currentNode.nodeKey == null && key == null) {
+                    currentNode.nodeValue = value;
+                    break;
+                } else {
+                    if (currentNode.next == null) {
+                        currentNode.next = new Node<>(key,value,null);
+                        size++;
+                        break;
+                    } else {
+                        currentNode = currentNode.next;
+                    }
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        Node<K, V> currentNode = values[getIndexFromKey(key)];
+        if (currentNode == null) {
+            return null;
+        }
+        while (currentNode != null) {
+            if (currentNode.nodeKey != null && currentNode.nodeKey.equals(key)
+                    || currentNode.nodeKey == key) {
+                return currentNode.nodeValue;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void arrayResize_Copy() {
```

Comment:
we don't use snake_case in java
```suggestion
    private void resize() {
```